packages/kbn-apm-synthtrace-client/src/lib/logs/index.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Fields } from '../entity';
+import { Serializable } from '../serializable';
+
+export type LogDocument = Fields &
+  Partial<{
```

Comment:
**nit:** What's the purpose of Partial here if the type is defined inline? You could remove it and directly mark all the properties as optional (most of them already are).

Reply:
I think Partial is useful here, because it makes it explicit that having the fields be partial is a functional requirement, and it's easy to forget an `?`.